Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,418 @@
+     *         client and Quadriga.
+     * @throws QuadrigaException
+     * @throws IOException
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws JAXBException
+     * @throws QuadrigaAccessException
+     * @throws QuadrigaStorageException
+     * @throws RestException
+     */
+    @RestAccessPolicies({
+            @ElementAccessPolicy(type = CheckedElementType.CONCEPTCOLLECTION_REST, paramIndex = 1, userRole = {
+                    RoleNames.ROLE_CC_COLLABORATOR_ADMIN, RoleNames.ROLE_CC_COLLABORATOR_READ_WRITE }) })
+    @RequestMapping(value = "rest/syncconcepts/{conceptCollectionID}", method = RequestMethod.POST)
+    public ResponseEntity<String> addConceptsToConceptColleciton(
+            @PathVariable("conceptCollectionID") String conceptCollectionId, HttpServletRequest request,
+            HttpServletResponse response, @RequestBody String xml, @RequestHeader("Accept") String accept,
+            Principal principal) throws QuadrigaException, ParserConfigurationException, SAXException, IOException,
+                    JAXBException, QuadrigaAccessException, QuadrigaStorageException, RestException {
+        IUser user = userManager.getUser(principal.getName());
+        if (xml.equals("")) {
+            String errorMsg = restMessage.getErrorMsg("Please provide XML in body of the post request.", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        } else {
+
+            logger.debug("XML : " + xml);
+            JAXBElement<QuadrigaConceptReply> response1 = null;
+            try {
+                JAXBContext context = JAXBContext.newInstance(QuadrigaConceptReply.class);
+                Unmarshaller unmarshaller = context.createUnmarshaller();
+                unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+                InputStream is = new ByteArrayInputStream(xml.getBytes());
+                response1 = unmarshaller.unmarshal(new StreamSource(is), QuadrigaConceptReply.class);
+            } catch (Exception e) {
+                logger.error("Error in unmarshalling", e);
+                String errorMsg = restMessage.getErrorMsg("Error in unmarshalling", request);
+                return new ResponseEntity<String>(errorMsg, HttpStatus.FORBIDDEN);
+            }
+            QuadrigaConceptReply qReply = response1.getValue();
+            ConceptList c1 = qReply.getConceptList();
+            List<Concept> conceptList = c1.getConcepts();
+
+            Iterator<Concept> I = conceptList.iterator();
+
+            while (I.hasNext()) {
+                Concept c = I.next();
+                logger.debug("arg Name :" + c.getName().trim());
+                logger.debug("arg Pos :" + c.getPos().trim());
+                logger.debug("arg URI :" + c.getUri().trim());
+                logger.debug("arg descrtiption :" + c.getDescription().trim());

```

Comment:
loose all these debug messages. if you really want to print the concept as debug msg, override the toString method and use that one instead. (one line of code)